prow/plugins/golint/suggestion/golint_suggestion.go

```diff
@@ -92,6 +94,34 @@ func fixStutter(p lint.Problem) string {
 	return suggestion
 }
 
+func fixErrorf(p lint.Problem) string {
+	matches := lintErrorfRegex.FindStringSubmatch(p.Text)
+	if len(matches) != 3 {
+		return ""
+	}
+	errorMessage := ""
+	count := 0
+	for _, char := range p.LineText[strings.Index(p.LineText, "errors.New(fmt.Sprintf(")+len("errors.New(fmt.Sprintf("):] {
+		if char == '(' {
+			count++
+		}
+		if char == ')' {
+			count--
+			if count < 0 {
+				break
+			}
+		}
+		errorMessage += string(char)
+	}
+	toReplace := strings.Replace(matches[1], "...", errorMessage, -1)
+	replacement := strings.Replace(matches[2], "...", errorMessage, -1)
```

Comment:
@cjwagner the regexp is used because the replacement `fmt.Errorf(...)` has one less closing parenthesis than the toReplace `errors.New(fmt.Sprintf(...))` so using `Replace` makes it easier to deal with the closing parenthesis. I'm not sure how using `fmt.Sprintf` would simplify this. Could you please clarify?

Reply:
```go
toReplace := fmt.Sprintf("errors.New(fmt.Sprintf(%s))", parameterText)
replacement := fmt.Sprintf("fmt.Errorf(%s)", parameterText)
```
I think this is simpler to read. It doesn't involve any regexp matching and clearly indicates what is being replaced and what the replacement is. The regexp matching doesn't make sense because you are replacing a constant from constant match (i.e. `match[1]` always has the same content and you are replacing the literal `...` so you mind as well just build the string that you want directly without the extra step of using a capture group from the match).

Reply:
Got it. Thanks for the clarification!